packages/gatsby/src/redux/reducers/index.js

```diff
@@ -41,4 +41,5 @@ module.exports = {
   babelrc: require(`./babelrc`),
   jsonDataPaths: require(`./json-data-paths`),
   thirdPartySchemas: require(`./thirdPartySchemas`),
+  themes: require(`./themes`),
```

Comment:
This would definitely solve (one of) the issues I was running into in #10786. If there's a JSON structure I can use that contains the known path to themes--that's all I need!

Still want to think about whether it might make sense to offer some mechanism _outside_ of themes (e.g. thinking like a custom component that uses StaticQuery or something), but that's not as crucial and could be a little fringe-y.

Reply:
A thought coming off of #10739: I had mentioned I believed that themes shouldn't need to supply a gatsby-config.js. I am going to walk back on that a bit. What if we consider the recursive composability of configs and the component shadowing as two separate pieces. Really, the configs could be useful within plugins as well (and I think you were coming to this as well @DSchau ). The component shadowing really seems to be specifically theme based.

Presuming themes will move into the main plugins array, we need to be able to identify them somehow. What if a theme needs to declare itself as a theme within the config (effectively requiring a config for themes)? Essentially it would be opting into the component shadowing as config composability is could be used by any plugin.

Thoughts?

Reply:
Mostly I think this conversation (What are the differences between themes and plugins and should they be defined in the same array) is best left to the time at which we start considering "Themes 1.0". There's plenty that will happen between now and then and IMO it's too early to optimize this. I'm intentionally kicking that conversation down the road as I believe there will be *more* functionality added to themes. It's far easier to later merge them into the same array if it turns out that's the right path than it is to split them up after we've merged them (which could result in more community turmoil and a longer deprecation process).